### PR TITLE
Added opportunity to set a custom icon on the menu button.

### PR DESCRIPTION
**Pull request by Poeschl**
I changed the menu class, so that you can set a own icon for the menu button. If the menu is expanded the icon transforms into a 'x' to indicate the close action.
Maybe this helps you with this issue: https://github.com/futuresimple/android-floating-action-button/issues/56

I also added the icon function to one of the menus in the sample. Feel free to give me any feedback about this.

### CODE REVIEW

**Comment by thesparky007** [comment]
Will this be merged into the main branch?

**Comment by Simon-Raes** [comment]
I encountered a small bug with this code where the collapsed drawable would be invisible if I closed and re-opened a screen while the menu was in the open state.

I fixed it by adding the following code [here](https://github.com/futuresimple/android-floating-action-button/pull/143/files#diff-cedd241c3089ff270ad464070a3463b1R200).
<code>                    layeredDrawable.getDrawable(COLLAPSED_DRAWABLE).setAlpha(MAX_ALPHA);
</code>
This resets the collapsed drawable's alpha to be visible again when opening the screen.

**Comment by ticofab** [comment]
Any news on merging this into the main branch?

**Comment by Poeschl** [comment]
@ticofab I'm not intending to push it to a Maven Repository. I was hoping to get this merged. And when I have some more time I will improve it to looks like the inbox button.

**Comment by pranayairan** [comment]
Will this be merged anytime soon ?? There are lot of great additions to the library that are waiting to be merged. @chalup

File: library/src/main/res/values/attrs.xml [hunk 1]
```diff
@@ -35,5 +35,6 @@
             <enum name="left" value="2"/>
             <enum name="right" value="3"/>
         </attr>
+        <attr name="fab_collapsedIcon" format="reference"/>>
```
Comment (luissiqueira) [inline library/src/main/res/values/attrs.xml]: />>

Comment (Poeschl) [inline library/src/main/res/values/attrs.xml]: Fixed in commit 0148bd7

**Comment by Zakay** [comment]
Changing the icon was the first thing I wanted to do. This should really be merged.

**Comment by Zakay** [comment]
@Poeschl I found a fork of this library which is more active and has more features.
https://github.com/Clans/FloatingActionButton

It has implemented custom icon for the menu button but the icon is used for both states and is rotated when clicked. One of the weirdest solutions.

If you add your fab_collapsedIcon to this project I think the author will merge it pretty quickly as he seems very active.

**Comment by Poeschl** [comment]
@Zakay Whuu, nice for that reply. If I find some time, I will do that and report here when I'm finished.

**Comment by Zakay** [comment]
I actually already sent a pull request!

**Comment by Poeschl** [comment]
Nice :)

**Comment by Poeschl** [comment]
@chalup Is there a chance to get this merged? It is laying around for nearly 30 days.

**Comment by ReckLink** [comment]
Any timeline for this getting merged to the main branch?  This feature is completely dependent on me using this library.

**Pull request closed without merging**